### PR TITLE
Add Response File Capabilities to XUnit Console Runner

### PR DESCRIPTION
**Pull request by josalem**
* consume a path prefixed by the `@` symbol
* file should be a space/tab delimited list of command line arguments
(may be on multiple lines)
* does not check file extension but typically it is `.rsp`

#1613 

CC - @sergiy-k

### CODE REVIEW

File: src/Microsoft.DotNet.XUnitConsoleRunner/src/CommandLine.cs [hunk 1]
```diff
@@ -109,13 +109,29 @@ static bool IsConfigFile(string fileName)
...
+        protected void ParseRspFile(string path)
+        {
+            if (!File.Exists(path))
+            {
+                throw new ArgumentException($"file not found: {path}");
+            }
+
+            foreach (string line in File.ReadLines(path))
+            {
+                foreach (var arg in line.Split(new char[] { ' ', '\t' }, StringSplitOptions.RemoveEmptyEntries).Reverse())
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.XUnitConsoleRunner/src/CommandLine.cs]: This does not work with paths that contain spaces. CommandLineUtils developed by @natemcmaster does this really well. Please take a look at his parser for the correct implementation: https://github.com/natemcmaster/CommandLineUtils/blob/f498cc7383b27730fd24486510573ab61ccab9d6/src/CommandLineUtils/Internal/ResponseFileParser.cs

**Comment by jonfortescue** [comment]
@josalem -- looks like the failures were not caused by your PR but by something that was already merged, so ignore what I said for now

File: src/Microsoft.DotNet.XUnitConsoleRunner/src/CommandLine.cs (hunk 1 above)
Comment (natemcmaster) [reply src/Microsoft.DotNet.XUnitConsoleRunner/src/CommandLine.cs]: +1 - I share @ViktorHofer's concern. Splitting on spaces alone is insufficient for response file parsing. How would a caller escape parameters with spaces in them?

**Comment by ViktorHofer** [comment]
I know that there are multiple modes in @natemcmaster's CommandLineUtils. Which mode did you port? The line one? Do we also need the other one?

**Comment by josalem** [comment]
>I know that there are multiple modes in @natemcmaster's CommandLineUtils rsp implementation. Which mode did you port? The line one? Do we also need the other one?

I ported the "space delimited" version, since it is more flexible.  It reads the file line-by-line _and_ parses each line so any combination of new-line delimited and space delimited args should work, e.g.,
```
-flag "arg \" with quotes and \' spaces" -otherFlag "otherFlag argument"
```
and
```
-flag
"arg \" with quotes and \' spaces"
-otherFlag
"otherFlag argument"
```
so I didn't see the point in including both impls.

**Comment by ViktorHofer** [comment]
LGTM if you have tested it locally and it works fine. Thanks for doing this!

**Pull request merged**